packages/kyc-oracle/KycOracle.js

```diff
@@ -0,0 +1,231 @@
+            this.whitelist.approveUser(userAddress, true, { gasLimit: gasEstimate, gasPrice: this.gasPrice });
+          }
+        }
+
+        return res.status(200).send(data);
+      } catch (err) {
+        console.log(err);
+        return res.status(500).send(err);
+      }
+    });
+
+    this.server = this.app.listen(port, () => {
+      console.log(`⭐️ KycOracle running on port ${this.server.address().port}`);
+    });
+  }
+
+  /**
+   * Initialises the Kyc Oracle so that it knows where to find the `Whitelist` contract
+   * @param  {string}  whitelistAddress The address of the current `Whitelist` contract
+   * @return {Promise}
+   */
+  async initialise(whitelistAddress) {
+    const network = await this.provider.getNetwork();
+    this.chainId = network.chainId;
+
+    this.whitelistContractDef = await this.loader.load({ contractName: "Whitelist" }, { abi: true, address: false });
+    this.whitelist = new ethers.Contract(whitelistAddress, this.whitelistContractDef.abi, this.wallet);
+
+    await this.updateGasEstimate("safeLow");
+    await this.createDB();
+  }
+
+  /**
+   * Update the gas estimate
+   * @param  {string}  Transaction speed (fastest, fast, safeLow)
+   * @return {Promise}
+   */
+  async updateGasEstimate(type) {
+    if (this.chainId === 100) {
+      this.gasPrice = ethers.utils.hexlify(1000000000);
+      return;
+    }
+
+    try {
+      // Get latest from ethGasStation
+      const { data } = await axios.get("https://ethgasstation.info/json/ethgasAPI.json");
+
+      if (data[type]) {
+        this.gasPrice = ethers.utils.hexlify((data[type] / 10) * 1e9);
+      } else {
+        this.gasPrice = ethers.utils.hexlify(20000000000);
+      }
+    } catch (err) {
+      console.log(`Error during gas estimation: ${err}`);
+      this.gasPrice = ethers.utils.hexlify(20000000000);
+    }
+  }
+
+  async createDB() {
+    const db = await sqlite.open({ filename: this.dbPath, driver: sqlite3.Database });
+    await db.run("CREATE TABLE IF NOT EXISTS users ( address text NOT NULL UNIQUE, session_id text NOT NULL  )");
+    await db.close();
+  }
+
+  async getSessionForAddress(address) {
+    const db = await sqlite.open({ filename: this.dbPath, driver: sqlite3.Database });
+    const res = await db.all(
+      `SELECT DISTINCT users.session_id as session_id
+       FROM users
+       WHERE users.address="${address}"`
+    );
+    await db.close();
+    const sessionIds = res.map((x) => x.session_id);
+    console.log(sessionIds);
+    if (sessionIds.length > 1) {
+      throw new Error("More than one matching address");
+    }
+    return sessionIds[0];
+  }
+
+  async setSessionForAddress(address, session) {
+    console.log(address, session);
+    const db = await sqlite.open({ filename: this.dbPath, driver: sqlite3.Database });
+    await db.run(
+      `INSERT INTO users (address, session_id) VALUES ('${address}', '${session}') ON CONFLICT(address) DO UPDATE SET session_id = '${session}'`
```

Comment:
Stylistically you're using single quotes here and double quotes earlier in the SQL queries.